utils/channel_access.py

```diff
@@ -7,36 +7,36 @@
 import operator
 import os
 import time
+from abc import abstractmethod
 from contextlib import contextmanager
-from functools import partial
+from functools import partial, partialmethod
+from typing import Callable, Generator, Optional
 
-from genie_python.genie_cachannel_wrapper import CaChannelWrapper, UnableToConnectToPVException
+from genie_python.genie import PVValue  # type: ignore
+from genie_python.genie_cachannel_wrapper import (  # type: ignore
+    CaChannelWrapper,
+    UnableToConnectToPVException,
+)
+from genie_python.genie_p4p_wrapper import P4PWrapper  # type: ignore
```

Comment:
btw, you should now be able to add `genie_python>=15.1.0-rc1` to `requirements.txt` and pyright on the build server should then pick everything up correctly 🤞 - i.e. you should hopefully not need the `type: ignore` any more...